base/reflection.jl

```diff
@@ -1107,6 +1107,33 @@ function visit(f, d::Core.TypeMapEntry)
     end
     nothing
 end
+struct MethodSpecializations
+    specializations::Union{Nothing, Core.MethodInstance, Core.SimpleVector}
+end
+"""
+    specializations(m::Method) → itr
+
+Return an iterator `itr` of all compiler-generated specializations of `m`.
+"""
+specializations(m::Method) = MethodSpecializations(isdefined(m, :specializations) ? m.specializations : nothing)
+function iterate(specs::MethodSpecializations)
+    s = specs.specializations
+    s === nothing && return nothing
+    isa(s, Core.MethodInstance) && return (s, false)
+    return iterate(specs, 0)
+end
+iterate(specs::MethodSpecializations, ::Bool) = nothing
+function iterate(specs::MethodSpecializations, i::Int)
+    s = specs.specializations
+    n = length(s)
+    i >= n && return nothing
+    item = nothing
+    while i < n && item === nothing
+        item = s[i+=1]
+    end
+    item === nothing && return nothing
+    return (item, i)
+end
 
```

Comment:
```suggestion
function length(specs::MethodSpecializations)
    s = specs.specializations
    if s === nothing
        return 0
    elseif s isa Core.MethodInstance
        return 1
    elseif s isa Core.SimpleVector
        return length(s)
    else
        @assert false
    end
end
```

Reply:
The issue is that there may not `nothing` slots in the SimpleVector. I'll define `length` but it will be implemented differently.